components/proxy/src/main/java/com/hotels/styx/StyxServer.java

```diff
@@ -163,45 +167,86 @@ public StyxServer(StyxServerComponents components, Stopwatch stopwatch) {
 
         registerCoreMetrics(components.environment().buildInfo(), components.environment().metricRegistry());
 
-        ProxyServerSetUp proxyServerSetUp = new ProxyServerSetUp(
-                new StyxPipelineFactory(
-                        components.routingObjectFactoryContext(),
-                        components.environment(),
-                        components.services(),
-                        components.plugins(),
-                        components.eventLoopGroup(),
-                        components.nettySocketChannelClass()));
-
-        this.proxyServer = proxyServerSetUp.createProxyServer(components);
-        this.adminServer = createAdminServer(components);
-
-        this.serviceManager = new ServiceManager(new ArrayList<Service>() {
-            {
-                add(proxyServer);
-                add(adminServer);
-
-                concat(components.services().values(),
-                        new ServiceProviderMonitor("Styx-Service-Monitor", components.servicesDatabase()))
-                    .map(StyxServer::toGuavaService)
-                    .forEach(this::add);
-            }
-        });
+        // The plugins are loaded, but not initialised.
+        // And therefore not able to accept traffic.
+        HttpHandler httpHandler = new StyxPipelineFactory(
+                components.routingObjectFactoryContext(),
+                components.environment(),
+                components.services(),
+                components.plugins(),
+                components.eventLoopGroup(),
+                components.nettySocketChannelClass())
+                .create();
+
+        // Startup phase 1: start plugins, control plane providers, and other services:
```

Comment:
Can we break up this method?

Reply:
Not really, because this is in a constructor, and assigning to final variables. 
But I will be returning to this soon. Expect more changes, but in separate increments.